src/main/java/org/folio/edge/sip2/repositories/CirculationRepository.java

```diff
@@ -109,71 +107,143 @@ public Future<CheckinResponse> performCheckinCommand(Checkin checkin, SessionDat
-                    .transactionDate(OffsetDateTime.now(clock))
-                    .institutionId(institutionId)
-                    .patronIdentifier(patronIdentifier)
-                    .itemIdentifier(itemIdentifier)
-                    .titleIdentifier(response.map(
-                        v -> getChildString(v, "item", "title", UNKNOWN)).orElse(UNKNOWN))
-                    .dueDate(dueDate)
-                    .screenMessage(Optional.of(resource.getErrorMessages())
-                        .filter(v -> !v.isEmpty())
-                        .orElse(null))
-                    .build();
-              });
-        });
+      .compose(verification -> {
+        if (FALSE.equals(verification.getPasswordVerified())) {
+          return Future.succeededFuture(CheckoutResponse.builder()
+            .ok(FALSE)
+            .renewalOk(FALSE)
+            .magneticMedia(null)
+            .desensitize(FALSE)
+            .transactionDate(OffsetDateTime.now(clock))
+            .institutionId(institutionId)
+            .patronIdentifier(patronIdentifier)
+            .itemIdentifier(itemIdentifier)
+            .titleIdentifier(UNKNOWN)
+            .dueDate(OffsetDateTime.now(clock))
+            .screenMessage(verification.getErrorMessages())
+            .build());
+        }
+
+        final User user = verification.getUser();
+        final JsonObject body = new JsonObject()
+          .put("itemBarcode", itemIdentifier)
+          .put("userBarcode", user != null ? user.getBarcode() : patronIdentifier)
+          .put("servicePointId", sessionData.getScLocation());
+
+        final Map<String, String> headers = getBaseHeaders();
+
+        final CheckoutRequestData checkoutRequestData =
+          new CheckoutRequestData(body, headers, sessionData);
+
+        final Future<IResource> result = resourceProvider.createResource(checkoutRequestData);
+
+        return result
+          .otherwise(Utils::handleErrors)
+          .compose(res -> addTitleIfNotFound(sessionData, itemIdentifier, res))
+          .map(resource -> {
+            final Optional<JsonObject> response = Optional.ofNullable(resource.getResource());
+
+            final OffsetDateTime dueDate = response
+              .map(v -> v.getString("dueDate"))
+              .map(v -> OffsetDateTime.from(Utils.getFolioDateTimeFormatter().parse(v)))
+              .orElse(OffsetDateTime.now(clock));
+
+            return CheckoutResponse.builder()
+              .ok(Boolean.valueOf(response.isPresent()))
+              .renewalOk(FALSE)
+              .magneticMedia(null)
+              .desensitize(Boolean.valueOf(response.isPresent()))
+              .transactionDate(OffsetDateTime.now(clock))
+              .institutionId(institutionId)
+              .patronIdentifier(patronIdentifier)
+              .itemIdentifier(itemIdentifier)
+              .titleIdentifier(response.map(
+                  v -> getChildString(v, "item", "title", UNKNOWN))
```

Comment:
I think we can remove this additional `getChildString` for `item` and `title` because we're now getting it above with `addTitleIfNotFound` and it will now be available in the resource. In other words this is now redundant. Also in my check of all the possible responses from the circ backend, I was unable to see JSON with `item` and `title`. See my [last comment](https://issues.folio.org/projects/SIP2/issues/SIP2-115) on the ticket for this research.

Then this can be simplified to `.titleIdentifier(resource.getTitle())`.

Reply:
I'm forgetting that this is needed to handle the non-error case.